### PR TITLE
chore: bump k8s to 1.28

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Updates k8s to v1.28 and corresponding solo-io dependencies

## API changes
- N/A

## Code changes
- Bump deps

## CI changes
- Bumped cloudbuilder
- Bumped kind cluster version
- Bumped k8s versions in test suites

## Docs changes
N/A

# Context
https://github.com/solo-io/gloo/issues/8515

## Interesting decisions
 
N/A

## Testing steps

CI

## Notes for reviewers

N/A

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: .github/workflows/nightly-tests.yaml [hunk 1]
```diff
@@ -45,8 +45,8 @@ jobs:
       fail-fast: false
       matrix:
         kube-e2e-test-type: ['gateway', 'gloo', 'ingress', 'helm', 'gloomtls', 'glooctl', 'upgrade']
-        kube-version: [ { node: 'v1.23.13@sha256:ef453bb7c79f0e3caba88d2067d4196f427794086a7d0df8df4f019d5e336b61', kubectl: 'v1.23.17', kind: 'v0.17.0', helm: 'v3.11.2' },
-                        { node: 'v1.27.3@sha256:3966ac761ae0136263ffdb6cfd4db23ef8a83cba8a463690e98317add2c9ba72', kubectl: 'v1.27.3', kind: 'v0.20.0', helm: 'v3.12.2' }]
+        kube-version: [ { node: 'v1.24.15@sha256:7db4f8bea3e14b82d12e044e25e34bd53754b7f2b0e9d56df21774e6f66a70ab', kubectl: 'v1.24.17', kind: 'v0.20.0', helm: 'v3.12.2' },
+                        { node: 'v1.28.0@sha256:b7a4cad12c197af3ba43202d3efe03246b3f0793f162afb40a33c923952d5b31', kubectl: 'v1.28.4', kind: 'v0.20.0', helm: 'v3.13.2' }]
```
[the remark this answers is not in the document]
Comment (davidjumani) [reply .github/workflows/nightly-tests.yaml:49]: The [latest tag](https://hub.docker.com/r/kindest/node/tags) for kind node images is v1.28.0

File: changelog/v1.16.0-beta28/bump-k8s-128.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  dependencyRepo: kubectl
+  dependencyTag: v0.28.3
+- type: DEPENDENCY_BUMP
+  dependencyOwner: sigs.k8s.io
+  dependencyRepo: controller-runtime
+  dependencyTag: v0.16.3
+- type: DEPENDENCY_BUMP
+  dependencyOwner: helm.sh
+  dependencyRepo: helm
+  dependencyTag: v3.13.2
```
[the remark this answers is not in the document]
Comment (davidjumani) [reply changelog/v1.16.0-beta28/bump-k8s-128.yaml:46]: Yes. Have mentioned it under other changes in the [original issue](https://github.com/solo-io/gloo/issues/8515#issuecomment-1824571571)

[the remark this answers is not in the document]
Comment (davidjumani) [reply changelog/v1.16.0-beta28/bump-k8s-128.yaml:46]: Yes. I've mentioned the need to update it in the [original issue](https://github.com/solo-io/gloo/issues/8515#issuecomment-1824571571)
Cloudbuilder has also been [bumped](https://github.com/solo-io/cloud-builders/pull/118) with this version of helm
```
- Helm upgrade to v3.13.2 - The lower version used kubectl v1.24 which is incompatible with k8s v1.28
- Moby upgrade to v24.0.7+incompatible - With the helm upgrade, the docker libs were bumped as well. Bumped moby to the corresponding version

```

**Review by nfuden — APPROVED** [review]

**Pull request merged**